tpot/base.py

```diff
@@ -353,6 +338,40 @@ def __init__(self, generations=100, population_size=100, offspring_size=None,
         self._setup_pset()
         self._setup_toolbox()
 
+
+    def _setup_scoring_function(self, scoring):
+        if scoring:
+            if isinstance(scoring, str):
+                if scoring not in SCORERS:
+                    raise ValueError(
+                        'The scoring function {} is not available. Please '
+                        'choose a valid scoring function from the TPOT '
+                        'documentation.'.format(scoring)
+                    )
+            elif callable(scoring):
+                if not isinstance(scoring, _BaseScorer):
+                    # If the user passed a custom metric function, store it in the sklearn
+                    # SCORERS dictionary
+                    scoring_name = scoring.__name__
+                    greater_is_better = 'loss' not in scoring_name and 'error' not in scoring_name
+                    SCORERS[scoring_name] = make_scorer(scoring, greater_is_better=greater_is_better)
+                    warnings.simplefilter('always', DeprecationWarning)
+                    warnings.warn('Scoring value {} looks like it is a metric function '
+                                    'rather than a scorer. This scoring type was deprecated '
+                                    'in version 0.9.1 and will be removed in version 0.11. '
+                                    'TPOT uses `make_scorer` to convert this metric function'
+                                    'to a scorer which requires an estimator as its first '
+                                    'parameter'.format(scoring), DeprecationWarning)
+
+                else:
+                    scoring_name = scoring._score_func.__name__
+                    SCORERS[scoring_name] = scoring
```

Comment:
@rhiever Here is the case of using board `SCORERS` dict for a scoring function with the signature `my_scorer(estimator, X, y)`

Reply:
OK, I see it now. Must've taken a minute to update.